dev-java/javassist/javassist-3.29.0.ebuild

```diff
@@ -0,0 +1,156 @@
+#	javassist.CannotCompileException: [source error] no such field: i
+#		at javassist.expr.NewArray.replace(NewArray.java:182)
+#		at javassist.JvstTest2$4.edit(JvstTest2.java:749)
+#		at javassist.expr.ExprEditor.loopBody(ExprEditor.java:238)
+#		at javassist.expr.ExprEditor.doit(ExprEditor.java:96)
+#		at javassist.CtBehavior.instrument(CtBehavior.java:744)
+#		at javassist.JvstTest2.testNewArray(JvstTest2.java:760)
+#		... 27 trimmed
+#	Caused by: compile error: no such field: i
+#		at javassist.compiler.TypeChecker.fieldAccess(TypeChecker.java:873)
+#		at javassist.compiler.TypeChecker.atFieldAssign(TypeChecker.java:296)
+#		at javassist.compiler.JvstTypeChecker.atFieldAssign(JvstTypeChecker.java:95)
+#		at javassist.compiler.TypeChecker.atAssignExpr(TypeChecker.java:255)
+#		at javassist.compiler.ast.AssignExpr.accept(AssignExpr.java:43)
+#		at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:266)
+#		at javassist.compiler.CodeGen.atStmnt(CodeGen.java:360)
+#		at javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
+#		at javassist.compiler.CodeGen.atStmnt(CodeGen.java:381)
+#		at javassist.compiler.ast.Stmnt.accept(Stmnt.java:53)
+#		at javassist.compiler.Javac.compileStmnt(Javac.java:578)
+#		at javassist.expr.NewArray.replace2(NewArray.java:252)
+#		at javassist.expr.NewArray.replace(NewArray.java:180)
+#		... 33 more
+#	4) testURL(javassist.JvstTest2)
+#	junit.framework.AssertionFailedError
+#		at junit.framework.Assert.fail(Assert.java:55)
+#		at junit.framework.Assert.assertTrue(Assert.java:22)
+#		at junit.framework.Assert.assertTrue(Assert.java:31)
+#		at junit.framework.TestCase.assertTrue(TestCase.java:200)
+#		at javassist.JvstTest2.testURL(JvstTest2.java:336)
+#	5) testMethodParameters(javassist.JvstTest4)
+#	junit.framework.ComparisonFailure: expected:<[i]> but was:<[arg0]>
+#		at junit.framework.Assert.assertEquals(Assert.java:100)
+#		at junit.framework.Assert.assertEquals(Assert.java:107)
+#		at junit.framework.TestCase.assertEquals(TestCase.java:260)
+#		at javassist.JvstTest4.testMethodParameters(JvstTest4.java:1010)
+#	6) testLocalVarAttribute(javassist.bytecode.BytecodeTest)
+#	junit.framework.AssertionFailedError
+#		at junit.framework.Assert.fail(Assert.java:55)
+#		at junit.framework.Assert.assertTrue(Assert.java:22)
+#		at junit.framework.Assert.assertTrue(Assert.java:31)
+#		at junit.framework.TestCase.assertTrue(TestCase.java:200)
+#		at javassist.bytecode.BytecodeTest.testLocalVarAttribute(BytecodeTest.java:335)
+#
+#	FAILURES!!!
+#	Tests run: 431,  Failures: 6
+# Cannot solve those test failures. Using "mvn test" they all pass.
+RESTRICT="test"
+
+# Compile dependencies
+# POM: pom.xml
+# test? junit:junit:[4.13.1,) -> >=dev-java/junit-4.13.2:4
+# test? org.hamcrest:hamcrest-all:1.3 -> !!!artifactId-not-found!!!
+
+DEPEND="
+	>=virtual/jdk-11:*
+	test? (
+		dev-java/hamcrest-library:1.3
+	)
+"
+
+# https://github.com/jboss-javassist/javassist/blob/rel_3_29_0_ga/pom.xml#L156-L157
+RDEPEND="
+	>=virtual/jre-11:*
```

Comment:
Isn't the code at the provided link in the comment just saying that the tests require java 11 while the library itself is fine with java 8? In that case, maybe we could make the it conditional on the 'test' use flag, whether or not we require at least java 8 or java 11?

Reply:
Tried that but failed.  Can you show me a working example?